packages/astro/src/core/build/plugins/plugin-manifest.ts

```diff
@@ -189,7 +189,10 @@ function buildManifest(
 	}
 
 	for (const route of opts.manifest.routes) {
-		const pageData = internals.pagesByComponent.get(route.component);
+		let pageData;
+		for (const page of internals.pagesByKeys.values()) {
+			if (page.route.route == route.route) pageData = page;
+		}
```

Comment:
```suggestion
		const key = `${route.route}_${route.component}`;
		const pageData = internals.pagesByKeys.get(key);
```

Maybe have a `computeRouteKey` or similarly named function to generate the key consistently everywhere it might be needed.